### PR TITLE
Add pct_change function for Series

### PR DESCRIPTION
**Pull request by TravisHester**
closes rapidsai/cudf#6133
unit test in test_stats.py: `test_series_pct_change`

Added `pct_change` function for Series. The `limit` and `freq` arguments could not be fully implemented due to lack of implementation in `Series.fillna()` and `Series.shift()` functions respectively.

### CODE REVIEW

File: python/cudf/cudf/core/series.py [hunk 1]
```diff
@@ -6019,6 +6019,44 @@ def explode(self, ignore_index=False):
...
+        Returns
+        -------
+        Series
+        """
+        if limit is not None:
+            raise NotImplementedError("limit parameter not supported yet.")
+        if freq is not None:
+            raise NotImplementedError("freq parameter not supported yet.")
+        if fill_method not in ["ffill", "bfill"]:
+            raise NotImplementedError(
```
Comment (isVoid) [inline python/cudf/cudf/core/series.py:6051]: Argument out of specified range is `ValueError`
```suggestion
            raise ValueError(
```

File: python/cudf/cudf/tests/test_stats.py [hunk 2]
```diff
@@ -316,6 +316,37 @@ def test_series_median(dtype, num_na):
...
+        got = cs.pct_change(periods=periods, fill_method=fill_method)
+        expected = ps.pct_change(periods=periods, fill_method=fill_method)
+        np.testing.assert_array_almost_equal(
+            got.to_array(fillna="pandas"), expected
+        )
+
+    with pytest.raises(NotImplementedError):
+        cs.pct_change(limit=1)
+    with pytest.raises(NotImplementedError):
+        cs.pct_change(freq="infer")
```
Comment (isVoid) [inline python/cudf/cudf/tests/test_stats.py:347]: We probably don't need to test for unsupported stuff... Especially this is run for every test case (redundant).

File: python/cudf/cudf/tests/test_stats.py [hunk 3]
```diff
@@ -316,6 +316,37 @@ def test_series_median(dtype, num_na):
...
+        cudf.Series([-3]),
+    ],
+)
+@pytest.mark.parametrize("periods", range(-5, 5))
+@pytest.mark.parametrize("fill_method", ["ffill", "bfill"])
+def test_series_pct_change(data, periods, fill_method):
+    cs = cudf.Series(data)
+    ps = cs.to_pandas()
+
+    if np.abs(periods) <= len(cs):
```
Comment (isVoid) [inline python/cudf/cudf/tests/test_stats.py:337]: This means the last two test cases `cudf.Series([])` and `cudf.Series([-3])` is untested.

File: python/cudf/cudf/tests/test_stats.py [hunk 4]
```diff
@@ -316,6 +316,37 @@ def test_series_median(dtype, num_na):
...
+        got = cs.pct_change(periods=periods, fill_method=fill_method)
+        expected = ps.pct_change(periods=periods, fill_method=fill_method)
+        np.testing.assert_array_almost_equal(
+            got.to_array(fillna="pandas"), expected
+        )
```
Comment (isVoid) [inline python/cudf/cudf/tests/test_stats.py:342]: I'm not sure about all the details here, but does `to_array(fillna="pandas")` really needed? This seems to introduce a preprocessing step before testing.

File: python/cudf/cudf/tests/test_stats.py (hunk 2 above)
Comment (TravisHester) [reply python/cudf/cudf/tests/test_stats.py:347]: Fair enough
```suggestion
```

File: python/cudf/cudf/tests/test_stats.py (hunk 4 above)
Comment (TravisHester) [reply python/cudf/cudf/tests/test_stats.py:342]: Yes, implicit conversion to a host NumPy array via `__array__` is not allowed. You have to use `to_array()` to compare them.

File: python/cudf/cudf/tests/test_stats.py (hunk 3 above)
Comment (TravisHester) [reply python/cudf/cudf/tests/test_stats.py:337]: The tested periods range from -5 to 5. The first will be tested when `periods` is equal to 0 and the second will be tested when `periods` is either -1, 0, or 1.

Comment (isVoid) [reply python/cudf/cudf/tests/test_stats.py:337]: Oops. Yup that's right.

**Review by isVoid — APPROVED** [review]

**Comment by beckernick** [comment]
@TravisHester this appears to be failing style checks

**Comment by beckernick** [comment]
Looks like some style changes + merge conflicts are all that remains

**Pull request closed without merging**